crates/bevy_image/src/image_loader.rs

```diff
@@ -12,6 +12,135 @@ pub struct ImageLoader {
 }
 
 impl ImageLoader {
+    /// Number of image formats, used for computing other constants.
+    const COUNT: usize = {
+        let mut count = 0;
+        #[cfg(feature = "avif")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "basis-universal")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "bmp")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "dds")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "ff")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "gif")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "ico")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "jpeg")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "ktx2")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "pnm")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "png")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "qoi")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "tga")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "tiff")]
+        {
+            count += 1;
+        }
+        #[cfg(feature = "webp")]
+        {
+            count += 1;
+        }
+        count
+    };
```

Comment:
```suggestion
    const COUNT: usize = Self::SUPPORTED_FORMATS.len();
```

Currently we have 1 feature flag for 1 format, so the length of `SUPPORTED_FORMATS` will be exactly `COUNT`.

Reply:
@bushrat011899 I think we should handle this in a separate PR, it's not clear to me why this can't be a loop that just sums all the file extension lengths. 

Regardless I'd like to keep the diff small, and `COUNT` is private so fixing it later won't affect our API.